Tasks/DockerV2/dockerpush.ts

```diff
@@ -79,13 +81,18 @@ export function run(connection: ContainerConnection, outputUpdate: (data: string
     let output = "";
     let outputImageName = "";
     let digest = "";
+    let imageSize = "";
     let promise = pushMultipleImages(connection, imageNames, tags, commandArguments, (image, commandOutput) => {
         output += commandOutput;
         outputImageName = image;
-        digest = extractDigestFromOutput(commandOutput);
-        tl.debug("outputImageName: " + outputImageName + "\n" + "commandOutput: " + commandOutput + "\n" + "digest:" + digest);
-        publishToImageMetadataStore(connection, outputImageName, tags, digest, dockerFile).then((result) => {
+        let extractedResult = extractDigestAndSizeFromOutput(commandOutput, matchPatternForDigestAndSize);
+        digest = extractedResult[0];
+        imageSize = extractedResult[1];
+        tl.debug("outputImageName: " + outputImageName + "\n" + "commandOutput: " + commandOutput + "\n" + "digest:" + digest + "imageSize:" + imageSize);
+        publishToImageMetadataStore(connection, outputImageName, tags, digest, dockerFile, imageSize).then((result) => {
             tl.debug("ImageDetailsApiResponse: " + result);
+        }, (error) => {
+            tl.debug("publishToImageMetadataStore failed with error: " + error);
```

Comment:
prefer warning if it doesnt impact task result so that we know in logs if this failed.